File: fixtures/scale/scala/medium/src/calc/memory/MemorySlot.scala
package calc.memory

final class MemorySlot {
  private var value = 0.0
  private var occupied = false

  def store(newValue: Double): Unit = {
    value = newValue
    occupied = true
  }

  def recall: Double = if (occupied) value else 0.0

  def clear(): Unit = {
    value = 0.0
    occupied = false
  }

  def isOccupied: Boolean = occupied
}

File: fixtures/scale/dart/medium/lib/memory/memory_slot.dart
class MemorySlot {
  double _value = 0.0;
  bool _occupied = false;

  void store(double newValue) {
    _value = newValue;
    _occupied = true;
  }

  double recall() => _occupied ? _value : 0.0;

  void clear() {
    _value = 0.0;
    _occupied = false;
  }

  bool get isOccupied => _occupied;
}
